remotes/mount.go

```diff
@@ -150,17 +177,70 @@ func (r *remoteReaderAt) ReadAt(p []byte, off int64) (int, error) {
 	return n, err
 }
 
-type descriptorAccumulator struct {
-	descriptors []ocischemav1.Descriptor
+type manifestWalker struct {
+	getChildren       images.HandlerFunc
+	resolver          docker.ResolverBlobMounter
+	targetRepo        string
+	eventNotifier     eventNotifier
+	descriptorHandler imageHandler
+	workQueue         *workQueue
+	progress          *progress
 }
 
-func (a *descriptorAccumulator) Handle(ctx context.Context, desc ocischemav1.Descriptor) ([]ocischemav1.Descriptor, error) {
-	descs := make([]ocischemav1.Descriptor, len(a.descriptors)+1)
-	descs[0] = desc
-	for i, d := range a.descriptors {
-		descs[i+1] = d
+func newManifestWalker(targetRepo string, resolver docker.ResolverBlobMounter, getChildren images.HandlerFunc,
+	descriptorHandler imageHandler, eventNotifier eventNotifier, workQueue *workQueue, progress *progress) *manifestWalker {
+	return &manifestWalker{
+		descriptorHandler: descriptorHandler,
+		eventNotifier:     eventNotifier,
+		getChildren:       getChildren,
+		resolver:          resolver,
+		targetRepo:        targetRepo,
+		workQueue:         workQueue,
+		progress:          progress,
 	}
+}
+
+var doneCh = func() chan struct{} {
+	ch := make(chan struct{})
+	close(ch)
+	return ch
+}()
 
-	a.descriptors = descs
-	return nil, nil
+func (w *manifestWalker) walk(ctx context.Context, desc ocischemav1.Descriptor, parent *descriptorProgress) (chan struct{}, error) {
+	descProgress := &descriptorProgress{
+		Descriptor: desc,
+	}
+	if parent != nil {
+		parent.addChild(descProgress)
+	} else {
+		w.progress.addRoot(descProgress)
+	}
+	copyOrMountWI := func(ctx context.Context) error {
```

Comment:
What does the `WI` suffix stand for here?

Reply:
WorkItem (will make it more obvious)